src/wallet/sigmaspendbuilder.cpp

```diff
@@ -168,13 +168,14 @@ CAmount SigmaSpendBuilder::GetChanges(std::vector<CTxOut>& outputs, CAmount amou
     auto params = sigma::Params::get_default();
 
     CHDMint hdMint;
+    CWalletDB walletdb(pwalletMain->strWalletFile);
```

Comment:
This is incorrect in multiple wallets scenario. You need to pass `CWalletDB` object reference to `GetChanges`

Reply:
Or you can obtain wallet file name from `CWallet` object and use it to create `CWalletDB`